agent.py

```diff
@@ -1031,13 +1033,11 @@ def findWelfare(self, sugarReward, spiceReward):
             totalSpice = 0
 
         welfare = (totalSugar ** sugarMetabolismProportion) * (totalSpice ** spiceMetabolismProportion)
-        if self.tags != None and len(self.tags) > 0:
+        if self.tagPreferences == True and self.tags != None and len(self.tags) > 0:
             # Tribe could have changed since last timestep, so recheck
             self.tribe = self.findTribe()
             fractionZeroesInTags = self.tagZeroes / len(self.tags)
             fractionOnesInTags = 1 - fractionZeroesInTags
-            spiceMetabolism = spiceMetabolism if spiceMetabolism > 0 else 1
-            sugarMetabolism = sugarMetabolism if sugarMetabolism > 0 else 1
             tagPreferences = (sugarMetabolism * fractionZeroesInTags) + (spiceMetabolism * fractionOnesInTags)
             if tagPreferences == 0:
```

Comment:
If removing lines 1039-1040, then this condition should be set to `<=` instead to prevent a negative preference since it's used in exponentiation.

Reply:
Good catch!